<launch>
  <arg name="launch_prefix"   default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace"  default="apriltag_ros_continuous_node" />
  <arg name="camera_name"     default="/wrist_camera/depth" />
  <arg name="image_topic"     default="image_raw" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find mobile_manip_demo)/config/tag_detection.yaml" ns="$(arg node_namespace)" />

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <!-- <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" /> -->
    <remap from="image_rect" to="/wrist_camera/color/image_raw" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>

  <node pkg="mobile_manip_demo" type="marker_detection.py" name="marker_detection_node" ns="$(arg node_namespace)" output="screen" />
</launch>
